data-raw/covid_pmfs.R

```diff
@@ -1,9 +1,116 @@
+  delay <- floor(secondary) - floor(primary)
+  if (missing(max)) {
+    max <- base::max(delay)
+  }
+  cdf <- ecdf(delay)(0:max)
+  pmf <- c(cdf[1], diff(cdf))
+  vec_outside_tol <- abs(sum(pmf) - 1L) > 1e-10
+  while (vec_outside_tol) {
+    pmf <- pmf / sum(pmf)
+    vec_outside_tol <- abs(sum(pmf) - 1L) > 1e-10
+  }
+  return(pmf)
+}
+
+
+# This retruns an incubation period pmf corresponding to the incubation period
+# for COVID after Omicron used in
+# Park et al 2023. These estimates are from early Omicron.
+
+make_incubation_period_pmf <- function(backward_scale = 3.60,
+                                       backward_shape = 1.50,
+                                       r = 0.15) {
+  # From: Park, Sang Woo, et al. "Inferring the differences in incubation-period
+  # and generation-interval distributions of the Delta and Omicron variants of
+  # SARS-CoV-2." Proceedings of the National Academy of Sciences 120.22 (2023):
+  # e2221887120.
+
+  # "However, when we account for growth-rate differences and reestimate the
+  # forward incubation periods, we find that both variants have similar
+  # incubation-period distributions with a mean of 4.1 d (95% CI: 3.8 to 4.4 d)
+  # for the Delta variant and 4.2 d (95% CI: 3.6 to 4.9 d) for the Omicron
+  # variant Fig. 3B)."
+
+  # Fits a Weibull to the data
+
+  # Relies on fundamental assumption about epidemic growth rate.
+
+
+  discr_gr_adj_weibull <- tibble::tibble(
+    time = seq(0, 23, by = 1), # 23 seems to get most of the distribution mass
+    density0 = dweibull(time,
+      shape = backward_shape,
+      scale = backward_scale
+    ) * exp(r * time)
+  )
+
+  inc_period_pmf <- to_simplex(discr_gr_adj_weibull$density0)
+  return(inc_period_pmf)
+}
+
+# Makes the hospital onset delay pmf using the parameter estimates based on
+# Danache et al linelist data from symptom onset to hospital
+# admission. See below:
+# https://journals.plos.org/plosone/article?id=10.1371/journal.pone.0261428
+
+make_hospital_onset_delay_pmf <- function(neg_binom_mu = 6.98665,
+                                          neg_binom_size = 2.490848) {
+  density <- dnbinom(
+    x = seq(0, 30, 1),
+    mu = neg_binom_mu, size = neg_binom_size
+  )
+  hosp_onset_delay_pmf <- density / sum(density)
+
+  return(hosp_onset_delay_pmf)
+}
+
+
+# Makes the reporting delay pmf by convolving the incubation period pmf with
+# the symptom to hospital admission pmf and normalizing
+
+make_reporting_delay_pmf <- function(incubation_period_pmf,
+                                     hospital_onset_delay_pmf) {
+  pmfs <- list(
+    "incubation_period" = incubation_period_pmf,
+    "hosp_onset_delay" = hospital_onset_delay_pmf
+  )
+
+  infection_to_hosp_delay_pmf <- add_pmfs(pmfs) |>
+    to_simplex()
+  return(infection_to_hosp_delay_pmf)
+}
+
+
+
+
+
+# Put it all together
 generation_interval <- withr::with_seed(42, {
   wwinference::simulate_double_censored_pmf(
```

Comment:
these are no longer package functions; they're defined above.

```suggestion
  simulate_double_censored_pmf(
```

Reply:
Same with the other calls below.